src/NUnitConsole/nunit3-console/TestEventHandler.cs

```diff
@@ -171,11 +184,25 @@ private void WriteOutputLine(string text, ColorStyle color)
             {
                 _outWriter.Write(text);
 
-                // Some labels were being shown on the same line as the previous output
+                // If the text we just wrote did not have a new line, flag that we should eventually emit one.
                 if (!text.EndsWith("\n"))
                 {
+                    _wantNewLine = true;
+                }
+            }
+        }
+
+        private void FlushNewLineIfNecessary(string testName)
+        {
+            if (_currentTestName != testName)
+            {
+                if (_wantNewLine)
+                {
                     _outWriter.WriteLine();
+                    _wantNewLine = false;
                 }
+
+                _currentTestName = testName;
```

Comment:
Can this be incorporated with or called from the methods that actually produce output?